python/tank/util/filesystem.py

```diff
@@ -343,3 +343,58 @@ def create_valid_filename(value):
         # so that we return a string
         u_src = value.decode("utf-8")
         return exp.sub("_", u_src).encode("utf-8")
+
+def get_permissions(path):
+    """
+    Retrieve the file system permissions for the file or folder in the
+    given path.
+
+    :param filename: Path to the file to be queried for permissions
+    :returns: permissions bits of the file 
+    :raises: OSError - if there was a problem retrieving permissions for the path
+    """
+    return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
+
+_delete_folder_success = True # global for function below
```

Comment:
Hey Luis, I guess this is my fault, I should have explained this better. `global` refers to any scope, not just the global scope. Here's what I had in mind. I'm omitting comments for brevity.

```
def delete_folder(path):
   delete_folder_success = True
   def _on_rm_error(func, path, exc_info):
       # This tells Python that assignments to delete_folder_success go to a variable defined outside this scope
        global delete_folder_success
        ...
 ```